### PR TITLE
EIP1-3439 / EIP1-3776 - Use real Explainer Documents

### PR DESCRIPTION
**Pull request by NathanRussellValtech**
This PR replaces the previous Temporary Certificate Explainer Document templates (our dev versions) with the real versions supplied by Malcolm Seaborn (UX / UCD)

The resultant PDFs with the fields populated look like this:

### English
Screenshot 2023-02-09 at 19 13 06

### Welsh
Screenshot 2023-02-09 at 19 13 29

### CODE REVIEW

File: src/main/resources/application.yml [hunk 1]
```diff
@@ -91,25 +91,25 @@ temporary-certificate:
     english:
       path: "classpath:temporary-certificate-template/Explainer Document (English).pdf"
       placeholder:
-        ero-name: "eroName"
```
Comment (NathanRussellValtech) [inline src/main/resources/application.yml:94]: The field names in the template PDFs have changed (unsurprisingly)

File: src/test/resources/application-integration-test.yml [hunk 2]
```diff
@@ -22,6 +22,9 @@ spring:
         format_sql : false
         show_sql : false
 
+  codec:
```
Comment (NathanRussellValtech) [inline src/test/resources/application-integration-test.yml:25]: We need this new property in the integration test spring context (`spring.codec.max-in-memory-size`) to allow the `WebTestClient` that is used in the integration tests to handle a large response body
(we have an integration test that generates and downloads the explainer doc - now we have the real explainer document template PDFs, the resultant http response is larger (almost 1MB !!)))

File: src/main/kotlin/uk/gov/dluhc/printapi/service/temporarycertificate/PdfFactory.kt [hunk 3]
```diff
@@ -16,6 +16,7 @@ class PdfFactory {
         PdfReader(ResourceUtils.getFile(templateDetails.path).inputStream()).use { reader ->
             val stamper = PdfStamper(reader, outputStream)
             stamper.cleanMetadata()
+            stamper.setFormFlattening(true)
```
Comment (NathanRussellValtech) [inline src/main/kotlin/uk/gov/dluhc/printapi/service/temporarycertificate/PdfFactory.kt:19]: Flatten the form fields when the stamper is closed, meaning the resultant PDF has the field values, but they are no longer editable form fields (dont want the vc-admin or someone creating a PDF, then being able to edit it themselves afterwards!)

File: src/test/kotlin/uk/gov/dluhc/printapi/rest/GenerateTemporaryCertificateExplainerDocumentIntegrationTest.kt [hunk 4]
```diff
@@ -148,8 +147,5 @@ internal class GenerateTemporaryCertificateExplainerDocumentIntegrationTest : In
 
         val pdfContent = result.responseBody
         assertThat(pdfContent).isNotNull
-        PdfReader(pdfContent).use { reader ->
```
Comment (NathanRussellValtech) [inline src/test/kotlin/uk/gov/dluhc/printapi/rest/GenerateTemporaryCertificateExplainerDocumentIntegrationTest.kt:151]: The resultant PDF that is generated and returned from the REST API no longer has form fields

File: src/test/resources/application-integration-test.yml (hunk 2 above)
Comment (NathanRussellValtech) [reply src/test/resources/application-integration-test.yml:25]: I would agree that your approach is more explicit. The property I have used is not exactly self-explanatory !!
I'll change it to use your approach 👍

Comment (NathanRussellValtech) [reply src/test/resources/application-integration-test.yml]: Done

File: src/main/resources/application.yml (hunk 1 above)
Comment (Neil-Massey) [reply src/main/resources/application.yml:94]: It's almost as if we knew that would happen and went to the effort of externalising their names 😄

File: src/main/kotlin/uk/gov/dluhc/printapi/service/temporarycertificate/PdfFactory.kt (hunk 3 above)
Comment (Neil-Massey) [reply src/main/kotlin/uk/gov/dluhc/printapi/service/temporarycertificate/PdfFactory.kt:19]: Nice - wasn't aware of this feature; have updated my PR to include this change 👍

Comment (matt-wills-valtech) [reply src/main/kotlin/uk/gov/dluhc/printapi/service/temporarycertificate/PdfFactory.kt:19]: Minor - perhaps add a comment to that effect?

**Review by matt-wills-valtech — APPROVED** [review]

**Review by adacayi — APPROVED** [review]

**Review by Neil-Massey — APPROVED** [review]
LGTM 👍

**Pull request merged**